### PR TITLE
Use CompletionStage instead of blocking in BaseSync and CategoryService

### PR DESCRIPTION
**Pull request by heshamMassoud**
Addresses issue #22:
1.  Use CompletionStage instead of blocking in BaseSync and CategoryService...
2. Gracefully handle Futures containing Exceptions.
3. Add some missing error messages as enums.

### CODE REVIEW

File: src/main/java/com/commercetools/sync/categories/CategorySync.java [hunk 1]
```diff
@@ -89,12 +96,13 @@ protected void processDrafts(@Nonnull final List<CategoryDraft> categoryDrafts)
...
         }
+        return CompletableFuture.completedFuture(this.statistics);
     }
 
     @Override
-    protected void process(@Nonnull final List<Category> resources) {
+    protected CompletionStage<CategorySyncStatistics> process(@Nonnull final List<Category> resources) {
         //TODO: SEE GITHUB ISSUE#12
-
+        return CompletableFuture.completedFuture(this.statistics);
```
Comment (jakubchrobak) [inline src/main/java/com/commercetools/sync/categories/CategorySync.java]: The `CategorySync` would be adjusted further in another PR right?

Comment (heshamMassoud) [inline src/main/java/com/commercetools/sync/categories/CategorySync.java]: yes

File: src/main/java/com/commercetools/sync/services/CategoryService.java [hunk 2]
```diff
@@ -7,16 +7,51 @@
...
-    Category createCategory(@Nonnull final CategoryDraft categoryDraft);
+    /**
+     * Given a {@link CategoryDraft}, this method creates a {@link Category} based on it in the CTP project defined in
+     * a potentially injected {@link io.sphere.sdk.client.SphereClient}. This method returns
+     * {@link CompletionStage&lt;Category&gt;} in which the result of it's completion contains an instance of the
+     * {@link Category} which was created in the CTP project.
+     *
+     * @param categoryDraft the {@link CategoryDraft} to create a {@link Category} based off of.
+     * @return {@link CompletionStage&lt;Category&gt;}containing as a result of it's completion an instance of the
+     *      {@link Category} which was created in the CTP project.
```
Comment (heshamMassoud) [inline src/main/java/com/commercetools/sync/services/CategoryService.java]: 0dec077

File: src/main/java/com/commercetools/sync/commons/enums/Error.java [hunk 3]
```diff
@@ -4,15 +4,26 @@
 import javax.annotation.Nonnull;
 
 /**
- * Represents all error messages across the sync update action build modules.
+ * Represents all error messages across the sync modules.
  */
 public enum Error {
```
Comment (jakubchrobak) [inline src/main/java/com/commercetools/sync/commons/enums/Error.java]: What is the gain from having all update messages in one place as public enum anyway? They are used always as ENUM.getDescritpion() inside format for calllback message. Why they're not kept as `private final static String` in context they belong to? Am I missing something?

**Comment by heshamMassoud** [comment]
@andrii-kovalenko-ct if you have time, can you please take a look at this PR.

File: src/main/java/com/commercetools/sync/commons/enums/Error.java (hunk 3 above)
Comment (heshamMassoud) [inline src/main/java/com/commercetools/sync/commons/enums/Error.java]: Actually, when I come to think of it again, it's actually better to use constant `Strings` instead of Enums in this case. Because most of them are _format_ `Strings` and if they are used as enums (`ENUM.getDescription()`) then the IDE will not be able to show warnings if the number of arguments don't match as opposed to directly using the constant String. 

Also it's not the best idea to keep all strings in one class (enum in this case), so if it get's really big and confusing. So I refactored them to be at the Utils where they are only needed. 	

29a1098
